cdap-tms/src/main/java/co/cask/cdap/messaging/subscriber/AbstractMessagingSubscriberService.java

```diff
@@ -127,35 +128,30 @@ protected void postProcess() {
   @Nullable
   @Override
   protected final String loadMessageId() {
-    return TransactionRunners.run(getTransactionRunner(), context -> {
-      return loadMessageId(context);
-    });
+    return TransactionRunners.run(getTransactionRunner(), (TxCallable<String>) this::loadMessageId);
   }
 
   @Nullable
   @Override
-  protected String processMessages(Iterator<ImmutablePair<String, T>> messages) {
+  protected String processMessages(Iterator<ImmutablePair<String, T>> messages) throws Exception {
```

Comment:
I think this class is also used to by other subscriber classes. If the loop is removed, all the other subscriber classes will not retry right?

Reply:
I removed the loop because it never loops. It always either returns or breaks.